lib/src/combined_wrappers/combined_list.dart

```diff
@@ -10,8 +10,13 @@ import 'dart:collection';
 /// concatenated list, but the underlying implementation is based on lazily
 /// accessing individual list instances.
 ///
-/// The returned list is unmodifiable.
+/// The resulting list has an index operator (`[]`) and `length` property that
+/// are both `O(lists)`, rather than `O(1)`, and the list is unmodifiable.
 List/*<T>*/ combineLists/*<T>*/(List<List/*<T>*/> lists) {
+  // Small optimization when there are no lists to avoid allocation.
+  if (lists.isEmpty) {
```

Comment:
Done, though without the closure allocation.

Also added an optimization for a single list, which is probably common enough too.